input/resources/au-core-medicationrequest.xml

```diff
@@ -248,7 +248,7 @@
         <key value="au-core-medreq-03"/>
         <severity value="error"/>
         <human value="Date shall be precise to the day or, if not available, the Data Absent Reason extension shall be present"/>
-        <expression value="(toString().length() &gt;= 8) xor extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()"/>
+        <expression value="($this is DateTime implies $this.toString().length() &gt;= 10) xor extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()"/>
```

Comment:
This still allows a non-precise date & a data absent reason, i.e.:
<authoredOn value="2020-07">
    <extension url="http://hl7.org/fhir/StructureDefinition/data-absent-reason">
      <valueCode value="masked"/>
    </extension>
</authoredOn>

Does not trigger the failure case for this invariant.

Reply:
Amended, thanks.

Reply:
No need for this complexity. Two simpler options are:

`$this is DateTime implies $this.toString().length() &gt;= 10 xor extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()`

or 

`$this.where($this.toString().length() &gt;= 10).exists() xor extension('
[http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()`

Reply:
Thanks for the suggestions. I have tested both and have included results below - neither was successful. I have kept the complex invariant.

$this is DateTime implies $this.toString().length() &gt;= 10 xor extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()
|Test|Outcome|
| --- | ---| 
|Valid DAR|pass|
|Valid date (to day)|pass|
|Valid date (to day) + DAR|fail|
|Valid date (to seconds)|pass|
|Valid date (to seconds) + DAR|fail|
|Invalid date (to month)|fail| 
|Invalid date (to month) + DAR|**pass**| 

$this.where($this.toString().length() &gt;= 10).exists() xor extension(' [http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()
|Test|Outcome|
| --- | ---| 
|Valid DAR|**fail**|
|Valid date (to day)|pass|
|Valid date (to day) + DAR|**pass**|
|Valid date (to seconds)|pass|
|Valid date (to seconds) + DAR|**pass**|
|Invalid date (to month)|fail| 
|Invalid date (to month) + DAR|fail|

Reply:
@JayMurdoch, could you please test this expression? I think it works and is the simplest one I could get to work (I think). 
 
`<expression value="(value.exists() and value.toString().length() &gt;= 10 and extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists().not()) or (value.empty() and extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists())"/>`

Reply:
Test results below - all behaving as expected. Thank you!

|Test|Outcome|
| --- | ---| 
|Valid DAR|pass|
|Valid date (to day)|pass|
|Valid date (to day) + DAR|fail|
|Valid date (to seconds)|pass|
|Valid date (to seconds) + DAR|fail|
|Invalid date (to month)|fail| 
|Invalid date (to month) + DAR|fail|